sphinx/util/typing.py

```diff
@@ -276,30 +340,32 @@ def restify(cls: type | None, mode: str = 'fully-qualified-except-typing') -> st
 def stringify_annotation(
     annotation: Any,
     /,
-    mode: str = 'fully-qualified-except-typing',
+    mode: str | RenderMode = RenderMode.fully_qualified_except_typing,
 ) -> str:
     """Stringify type annotation object.
 
     :param annotation: The annotation to stringified.
     :param mode: Specify a method how annotations will be stringified.
 
-                 'fully-qualified-except-typing'
-                     Show the module name and qualified name of the annotation except
-                     the "typing" module.
-                 'smart'
-                     Show the name of the annotation.
-                 'fully-qualified'
-                     Show the module name and qualified name of the annotation.
+    The following values can be given as a shorthand of a rendering mode:
+
+    * ``smart`` -- :attr:`RenderMode.smart`.
+    * ``fully-qualified`` -- :attr:`RenderMode.fully_qualified`.
+    * ``fully-qualified-except-typing`` -- :attr:`RenderMode.fully_qualified_except_typing`.
     """
     from sphinx.ext.autodoc.mock import ismock, ismockmodule  # lazy loading
     from sphinx.util.inspect import isNewType  # lazy loading
 
-    if mode not in {'fully-qualified-except-typing', 'fully-qualified', 'smart'}:
+    if isinstance(mode, str) and mode not in {
+        'fully-qualified-except-typing', 'fully-qualified', 'smart',
+    }:
```

Comment:
Maybe better to relocate this validation logic into `_normalize_mode`?

Reply:
Nope, my mistake I think.  It raises `ValueError` already if it does not recognize a `str` mode.

Reply:
No because the normalization is in `autodoc` and this one is in another module. I don't want to imports modules that could pollute it.